datafusion/expr/src/utils.rs

```diff
@@ -984,4 +1120,48 @@ mod tests {
         assert_eq!(expected, result);
         Ok(())
     }
+
+    #[test]
+    fn avoid_generate_duplicate_sort_keys() -> Result<()> {
+        let asc_or_desc = [true, false];
+        let nulls_first_or_last = [true, false];
+        let partition_by = &[col("age"), col("name"), col("created_at")];
+        for asc_ in asc_or_desc {
+            for nulls_first_ in nulls_first_or_last {
+                let order_by = &[
+                    Expr::Sort {
+                        expr: Box::new(col("age")),
+                        asc: asc_,
+                        nulls_first: nulls_first_,
+                    },
+                    Expr::Sort {
+                        expr: Box::new(col("name")),
+                        asc: asc_,
+                        nulls_first: nulls_first_,
+                    },
+                ];
+
+                let expected = vec![
+                    Expr::Sort {
+                        expr: Box::new(col("age")),
+                        asc: asc_,
+                        nulls_first: nulls_first_,
+                    },
+                    Expr::Sort {
+                        expr: Box::new(col("name")),
+                        asc: asc_,
+                        nulls_first: nulls_first_,
+                    },
+                    Expr::Sort {
+                        expr: Box::new(col("created_at")),
+                        asc: true,
+                        nulls_first: false,
+                    },
+                ];
+                let result = generate_sort_key(partition_by, order_by)?;
+                assert_eq!(expected, result);
```

Comment:
I checked with this commit it will
change :
`[age ASC NULLS FIRST, name ASC NULLS FIRST, created_at ASC NULLS FIRST, age ASC NULLS LAST, name ASC NULLS LAST]`
to:
`[age DESC NULLS FIRST, name DESC NULLS FIRST, created_at ASC NULLS LAST]`
👍

Reply:
yes, this PR will do additional dedup to avoid generate duplicate sort keys.